### PR TITLE
Update Range mapping type docs

### PR DESCRIPTION
**Pull request by wylieconlon**
The documentation for this mapping type was missing some important information about how to define and use ranges.

### CODE REVIEW

File: docs/reference/mapping/types/range.asciidoc [hunk 1]
```diff
@@ -4,6 +4,13 @@
 <titleabbrev>Range</titleabbrev>
 ++++
 
+A field that represents a relation between a lower and upper bound, using the
```
Comment (cbuescher) [inline docs/reference/mapping/types/range.asciidoc:7]: nit: might be just me, but I feel like the field doesn't represent the "relation" between the bounds but a continuous range of values between those bounds. The relations are then used to define and query the range. Maybe my understanding is off here, not a native speaker, sorry.

File: docs/reference/mapping/types/range.asciidoc [hunk 2]
```diff
@@ -4,6 +4,13 @@
...
+relations `gt`, `gte`, `lt` and `lte`. They can be used for querying, and have
+limited support for aggregations. The only supported aggregations are
+{ref}/search-aggregations-bucket-histogram-aggregation.html[histogram],
+{ref}/search-aggregations-bucket-range-aggregation.html[range] and
```
Comment (cbuescher) [inline docs/reference/mapping/types/range.asciidoc:11]: Just tried `range` aggregation on simple `integer_range` and `date_range` fields on 7.7 and master, gettting errors for both that these fields are not supported for this type of field. Maybe I'm missing something, what did you check or refer to when adding this?

**Review by cbuescher — COMMENTED** [review]
Left a small nit and am not sure if "range" aggs are supported for this `range` fields, at least I didn't get it to work when I tried, but maybe I'm missing something. My active knowledge is limited here.

**Review by jrodewig — COMMENTED** [review]
I left some feedback around the information hierarchy and some suggestions for improvement.

Let me know if you'd like any help implementing those. Thanks for the PR!

**Review by jrodewig — APPROVED** [review]
👍  LGTM. Thanks for the iteration @wylieconlon.

I'll merge and backport when @cbuescher approves.

**Review by cbuescher — APPROVED** [review]
Thanks for the changes, LGTM

**Pull request merged**